core/src/main/java/nl/nn/adapterframework/webcontrol/api/BrowseJdbcTable.java

```diff
@@ -134,32 +136,58 @@ public Response execute(LinkedHashMap<String, Object> json) throws ApiException
 				ResultSet rs = null;
 				try {
 					rs = conn.getMetaData().getColumns(null, null, tableName, null);
+
 					if (!rs.isBeforeFirst()) {
 						rs.close();
 						rs = conn.getMetaData().getColumns(null, null, tableName.toUpperCase(), null);
 					}
 	
-					String fielddefinition = "<fielddefinition>";
-					while(rs.next()) {
-						String field = "<field name=\"" + rs.getString(4) + "\" type=\"" + DB2XMLWriter.getFieldType(rs.getInt(5)) + "\" size=\"" + rs.getInt(7) + "\"/>";
-						fielddefinition = fielddefinition + field;
-						fieldDef.put(rs.getString(4), DB2XMLWriter.getFieldType(rs.getInt(5)) + "("+rs.getInt(7)+")");
+					StringBuilder fielddefinition = new StringBuilder("<fielddefinition>");
+					String field = null;
+					if(!numberOfRowsOnly) {
+						field = "<field name=\""+rnumColumnName+"\" type=\"INTEGER\" />";
+						fielddefinition.append(field);
+						fieldDef.put(rnumColumnName, "INTEGER");
+						while(rs.next()) {
+							field = "<field name=\"" + rs.getString(4) + "\" type=\"" + DB2XMLWriter.getFieldType(rs.getInt(5)) + "\" size=\"" + rs.getInt(7) + "\"/>";
+							fielddefinition.append(field);
+							fieldDef.put(rs.getString(4), DB2XMLWriter.getFieldType(rs.getInt(5)) + "("+rs.getInt(7)+")");
+						}
```

Comment:
Would it be possibe to use descriptive constants or column names, instead of '4', '5' and '7'?